runtime/jcl/common/mgmtthread.c

```diff
@@ -1287,8 +1291,28 @@ freeThreadInfos(J9VMThread *currentThread, ThreadInfo *allinfo, UDATA allinfolen
 		while (sinfo) {
 			next = sinfo->next;
 			j9mem_free_memory(sinfo);
+			if (sinfo->obj.unsafe != NULL) {
+				vmfns->j9jni_deleteLocalRef((JNIEnv *)currentThread, (jobject) sinfo->obj.unsafe);
+			}
 			sinfo = next;
 		}
+		if(allinfo[i].thread != NULL) {
+			vmfns->j9jni_deleteLocalRef((JNIEnv *)currentThread, allinfo[i].thread);
+		}
+		if(allinfo[i].blocker!= NULL) {
+			vmfns->j9jni_deleteLocalRef((JNIEnv *)currentThread, allinfo[i].blocker);
+		}
+		if(allinfo[i].blockerOwner!= NULL) {
+			vmfns->j9jni_deleteLocalRef((JNIEnv *)currentThread, allinfo[i].blockerOwner);
+		}
+		if(allinfo[i].stackTrace != NULL) {
+			vmfns->j9jni_deleteLocalRef((JNIEnv *)currentThread, allinfo[i].stackTrace);
+		}
+		for (int j = 0; j < allinfo[i].lockedMonitors.len; ++j) {
```

Comment:
This is not legal C code; `j` must be declared at the beginning of a block.

Reply:
fixed.